parsers/nmparser/get_model_output.go

```diff
@@ -67,48 +68,137 @@ func GetModelOutput(filePath string, verbose bool, noExt bool, noGrd bool, noCov
 		}
 	}
 
+	epsCount := 0
+	for i := range results.ParameterStructures.Sigma {
+		if results.ParameterStructures.Sigma[i] > 0 {
+			epsCount++
+		}
+	}
+	etaCount := 0
+	for i := range results.ParameterStructures.Omega {
+		if results.ParameterStructures.Omega[i] > 0 {
+			etaCount++
+		}
+	}
+
+	if !noShk {
+		shkFilePath := strings.Join([]string{filepath.Join(dir, runNum), ".shk"}, "")
+		shkLines, err := utils.ReadLines(shkFilePath)
+		if err != nil {
+			panic(err)
+		}
+		results.ShrinkageDetails = ParseShkData(ParseShkLines(shkLines), etaCount, epsCount)
+	}
+
+	setDefaultValues(&results, etaCount, epsCount)
+	return results
+}
+
+func setDefaultValues(results *ModelOutput, etaCount, epsCount int) {
+	// method name
+	for i := range results.ParametersData {
+		if len(results.ParametersData[i].Method) == 0 {
+			results.ParametersData[i].Method = "METHOD NOT DETECTED"
+		}
+	}
+
+	// stderr
 	for i := range results.ParametersData {
-		if len(results.ParametersData[i].Estimates.Theta) != len(results.ParametersData[i].StdErr.Theta) {
-			results.ParametersData[i].StdErr.Theta = make([]float64, len(results.ParametersData[i].Estimates.Theta))
+		var nameCount int
+
+		thetaCount := len(results.ParametersData[i].Estimates.Theta)
+		omegaCount := len(results.ParametersData[i].Estimates.Omega)
+		sigmaCount := len(results.ParametersData[i].Estimates.Sigma)
+
+		// stderr
+		nameCount = len(results.ParametersData[i].StdErr.Theta)
+		for n := nameCount; n < thetaCount; n++ {
+			results.ParametersData[i].StdErr.Theta = append(results.ParametersData[i].StdErr.Theta, 0)
```

Comment:
I'm not sure what this is doing?

Reply:
this definitely needs some inline comments explaining the top level logic of whats going on here

Reply:
OK, I'll add comments to this effect:
if the theta "stderr" array has less values than the theta "estimates" array, pad with 0's.

wip to use default values instead of 0's.  BTW,  there are two kinds of defaults values:
1. slices, which can't be done until dimensions are known (somehow)
2. values, it's better to initialize these when the object is created and then allow the default to be overwritten

Reply:
comments have been added

Reply:
I don't have an exhaustive list of all of the use-cases in which NONMEM can fail. This is a general solution since we know the dimensions _should_ be equal. Here is an example from the TMDD lst file:


THETA - VECTOR OF FIXED EFFECTS PARAMETERS   *********


         TH 1      TH 2      TH 3      TH 4      TH 5      TH 6      TH 7      TH 8      TH 9     
 
         2.68E-01  2.17E+00  1.44E-01  1.19E+00  5.57E+01  1.96E+00  8.00E-02  1.50E-01  1.05E+01


THETA - VECTOR OF FIXED EFFECTS PARAMETERS   *********


         TH 1      TH 2      TH 3      TH 4      TH 5      TH 6      TH 7      TH 8      TH 9     
 
         1.57E-02  1.61E-01  5.51E-03  3.26E-02  3.71E+00  3.97E-02  1.65E-03 ......... .........